index.js

```diff
@@ -307,5 +307,93 @@ class Stack {
   }
 }
 
+class SinglyLinkedList {
+
+	/**
+	 * @What instantiates the list.
+	 */
+	constructor() {
+		this.size = 0;
+		this.head = null;
+    this.tail = null;
+  }
+  
+  /**
+   * This method is specific to the class.
+   * @What creates a newNode for the list. Sets the next as null.
+   * @Params {js object or a primitive datatype} data 
+   */
+  newListNode(data) {
+    let node = {
+      'data': data,
+      'next': null
+    }
+    return node;
+  }
+
+	/**
+	 * @what Gets singly LL size
+	 */
+	getSize() {
+		return this.size;
+	}
+
+	/**
+	 * @What Adds a node to the beginning of the list.
+	 * @Params {value of the node} val 
+	 */
+	addFirst(val) {
+		var node = this.newListNode(val);
+		if (this.head === null) {
+			this.head = node;
+			this.tail = this.head;
+		} else {
+			node.next = this.head;
+			this.head = node;
+		}
+
+		this.size += 1;
+	}
+
+	/**
+	 * @What Adds a node to the end of the list.
+	 * @Params {value of the node} val 
+	 */
+	addLast(val) {
+		var node = this.newListNode(val);
+		if (this.tail === null) {
+			this.tail = node;
+			this.head = this.tail;
+		} else {
+			this.tail.next = node;
+			this.tail = this.tail.next;
+		}
+
+		this.size += 1;
+	}
+
+	/**
+	 * @What Prints list on console
+	 */
+	print() {
+		var listStr = "[ ";
```

Comment:
- Please push the nodes into an Array and join using '->'. 

- Please do not use true as the condition in the while and add something like while(runner), thus you would not need to break the loop and just push it in the Array while traversing the list.